warp/warp_service.go

```diff
@@ -0,0 +1,26 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package warp
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+)
+
+// WarpAPI introduces snowman specific functionality to the evm
+type WarpAPI struct {
+	Backend WarpBackend
+}
+
+// GetSignature returns the BLS signature associated with a messageID.
+func (api *WarpAPI) GetSignature(ctx context.Context, messageID ids.ID) (hexutil.Bytes, error) {
```

Comment:
Just confirming that the `params` argument in the JSON request can be hex encoded with this signature. Similarly, can you confirm that returning `hexutil.Bytes` directly gives a hex encoded output?

Reply:
Testing locally with modified code to get a response:

```bash
curl --location --request POST 'http://127.0.0.1:9652/ext/bc/2ioeFdBb3qMEfnTQc9kd3CqXBmTmxqh96GyCY49b8X3UmZx5fZ/rpc' \
--header 'Content-Type: application/json' \
--data-raw '{
    "jsonrpc": "2.0",
    "method": "warp_getSignature",
    "params": [
        "11111111111111111111111111111111LpoYY"
    ],
    "id": 1
}'
```

Output:
```json
{
    "jsonrpc": "2.0",
    "id": 1,
    "result": "0x0000000000000000000000000000000000000000000000000000000000000000"
}
```

with a different response output:
```json
{
    "jsonrpc": "2.0",
    "id": 1,
    "result": "0x6d0d36349f470af56409c02b53d5dfcc1a78071961f456321d9c707489d7116f"
}
```

Reply:
Does `c.client.CallContext(ctx, &res, "warp_getSignature", messageID[:])` in `warpClient.GetSignature` encode the bytes as cb58? I had done that manually in https://github.com/ava-labs/subnet-evm/pull/475, but did not test if it did that cb58 coding automatically.

Reply:
Yes it needs to be `messageID` so the type has the json encoding attached.